crates/valgrind/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+use std::cmp::*;
+use std::collections::HashMap;
+
+pub struct Valgrind {
+    metadata: Vec<MemState>,
+    mallocs: HashMap<usize, usize>,
+    pub stack_pointer: usize,
+    max_stack_size: usize,
```

Comment:
To reduce ambiguity I think this should be called `stack_end` or something similar? ("Stack size" makes me think of the actual size, so a stack from 1MiB down to 1MiB - 4KiB is 4KiB large; but the "end" is 1MiB)